src/components/FileViewer/FileViewer.tsx

```diff
@@ -218,33 +221,54 @@ export default function FileViewer() {
     }
   };
 
-  const zipFiles = async (files: ClientFile[], dest : string) => {
+  const zipFiles = async (files: ClientFile[], dest: string) => {
     await zipInstanceFiles(instance.uuid, {
       target_relative_paths: files.map((f) => f.path),
       destination_relative_path: dest,
     });
   };
 
-  const unzipFile = async (file: ClientFile, unzipOption : UnzipOption)  => { 
+  const unzipFile = async (file: ClientFile, unzipOption: UnzipOption) => {
     if (file.file_type !== 'File') {
       toast.error('Only files can be unzipped');
       return;
     }
 
-    await unzipInstanceFile(
-      instance.uuid,
-      file,
-      unzipOption,
-    );
-  }
+    await unzipInstanceFile(instance.uuid, file, unzipOption);
+  };
 
   const unzipTickedFile = async () => {
     if (!tickedFiles) return;
     const file = tickedFiles[0];
-    await unzipFile(file, "Smart");
+    await unzipFile(file, 'Smart');
     tickFile(file, false);
   };
 
+  const handleFileDrop = async (event: React.DragEvent) => {
+    if (!event.dataTransfer.types.includes('Files'))
+      return;
+
+    for (const i in event.dataTransfer.items) {
+        const item = event.dataTransfer.items[i];
+        if (item.kind !== "file") continue;
+        const entry = "getAsEntry" in DataTransferItem.prototype ? (item as any).getAsEntry() : item.webkitGetAsEntry();
+        if (entry.isDirectory) {
+          toast.error("Only files can be uploaded");
+          setDropping(false);
+          setDroppingDialog(false);
+          return;
```

Comment:
might be better to `continue` than return and make this a warning?

Reply:
not entirely sure what causes the behavior but this `continue` is pretty much only reached when uploading a normal file. at the start of the for loop, it sometimes yields an item with an undefined `kind` which causes the `webkitGetAsEntry()` to not work. not sure how else to do this part